nwbinspector/checks/nwb_containers.py

```diff
@@ -1,26 +1,24 @@
 """Authors: Cody Baker, Ben Dichter, and Ryan Ly."""
-import numpy as np
-
 import h5py
-from pynwb import NWBContainer, TimeSeries
+from pynwb import NWBContainer
 
-from ..utils import nwbinspector_check
+from ..utils import register_check
 
 
-@nwbinspector_check(severity=1, neurodata_type=NWBContainer)
-def check_dataset_compression(nwb_container: NWBContainer, bytes_threshold=2e6):
+@register_check(importance="Best Practice Violation", neurodata_type=NWBContainer)
```

Comment:
One way we could formalize this is to have in nwbinspector.py

```python
CRITICAL = 2
BEST_PRACTICE_VIOLATION = 1
BEST_PRACTICE_SUGGESTION = 0
```

and then here

```python
from .nwbinspector import (
    CRITICAL,
    BEST_PRACTICE_VIOLATION,
    BEST_PRACTICE_SUGGESTION,
)
...
@register_check(importance=BEST_PRACTICE_VIOLATION, neurodata_type=NWBContainer)
```

This would have the advantage that it would communicate to the developers what categories are available, and would allow for code auto-completion and inspection that would prevent typos.

Reply:
Oh, that's a good point on IDE level stuff. And string format checking, which can be a hassle.